packages/preferences/src/browser/preference-provider-user.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { ResourceProvider, Resource } from "@theia/core/lib/common";
+import { ILogger } from "@theia/core/lib/common";
+import URI from '@theia/core/lib/common/uri';
+import { DisposableCollection } from '@theia/core/lib/common';
+import { PreferenceProvider, PreferenceProviderClient, PreferenceProviderPriority } from '@theia/preferences-api/lib/browser';
+import * as jsoncparser from "jsonc-parser";
+import { ParseError } from "jsonc-parser";
+
+@injectable()
+export class PreferenceProviderUser implements PreferenceProvider {
```

Comment:
the naming looks awkward to me, `UserPreferenceProvider` and `WorkspacePreferenceProvider` is IMHO better